### PR TITLE
Framework: Introduce calypso-ui

### PR DESCRIPTION
**Pull request by ockham**
#### Changes proposed in this Pull Request

Create a `calypso-ui` package, and move `ScreenReaderText` and `ProgressBar` components there.

Alternative to @blowery's #32869 from which I've stolen pretty much all of the business logic. The only difference is that `ScreenReaderText` and `ProgressBar` are simpler components than `Dialog` and `RootChild` and thus easier to untangle.

#### Testing instructions

```
npm run distclean && npm ci
npm start
```

Does Calypso build and run?

### CODE REVIEW

**Comment by ockham** [comment]
Tests are failing since Jest gets confused by `import './style.scss'`. Need to tell it to ignore or mock those files.

My preferred solution would be [`transformIgnorePatterns`](https://jestjs.io/docs/en/configuration#transformignorepatterns-array-string) (probably in `calypso-build`'s `jest.config.js`), but I can't get that to work :tablefilp:

```diff
diff --git a/packages/calypso-build/jest.config.js b/packages/calypso-build/jest.config.js
index 8d2125240c..7efb5b7d78 100644
--- a/packages/calypso-build/jest.config.js
+++ b/packages/calypso-build/jest.config.js
@@ -2,5 +2,6 @@ module.exports = {
        testEnvironment: 'node',
        testMatch: [ '<rootDir>/**/test/*.js?(x)', '!**/.eslintrc.*' ],
        transform: { '^.+\\.jsx?$': 'babel-jest' },
+       transformIgnorePatterns: [ '^.+\\.scss?$' ],
        verbose: false,
 };
```

**Comment by ockham** [comment]
Relevant: https://github.com/Automattic/newspack-plugin/pull/48 :grimacing:

**Comment by justinshreve** [comment]
As a follow-up to your note here about aligning https://github.com/Automattic/newspack-plugin/pull/72#issuecomment-495187034, I was looking for some thoughts/feedback...

As part of the new WooCommerce Onboarding wizard, users are redirected into a special branded WooCommerce / Jetpack Connect flow that matches the rest of the core setup wizard. This includes using Muriel styled components throughout the experience. More specifically, the version of `TextControl` that aligns the label inside the input is used, which requires a bit more than just CSS to make work.

Screen Shot 2019-05-28 at 10 36 02 AM

We are using the first version of the Newspack components in WooCommerce Admin to avoid duplicating work which is great. We use a few different components from their package (checkbox, select, text input). 

However, it feels a bit odd to add Newspack as a dependency to Calypso, just for use in our special version of the sign-up flow. We also only need the single `TextControl` component on the Calypso side of this.

tl;dr: Should we consider porting that version of `TextControl` as another early `calypso-ui` component? I'm happy to take a look at this after our meetup next week if so. If not, are we OK pulling those components in temporarily? Or should I just do the overrides I need in the context of our code with a note to migrate in the future?

**Comment by ockham** [comment]
Rebased

**Comment by ockham** [comment]
> As a follow-up to your note here about aligning [Automattic/newspack-plugin#72 (comment)](https://github.com/Automattic/newspack-plugin/pull/72#issuecomment-495187034), I was looking for some thoughts/feedback...
>
> As part of the new WooCommerce Onboarding wizard, users are redirected into a special branded WooCommerce / Jetpack Connect flow that matches the rest of the core setup wizard. This includes using Muriel styled components throughout the experience. More specifically, the version of `TextControl` that aligns the label inside the input is used, which requires a bit more than just CSS to make work.
> 
> Screen Shot 2019-05-28 at 10 36 02 AM
> 
> We are using the first version of the Newspack components in WooCommerce Admin to avoid duplicating work which is great. We use a few different components from their package (checkbox, select, text input).
> 
> However, it feels a bit odd to add Newspack as a dependency to Calypso, just for use in our special version of the sign-up flow. We also only need the single `TextControl` component on the Calypso side of this.
> 
> tl;dr: Should we consider porting that version of `TextControl` as another early `calypso-ui` component? I'm happy to take a look at this after our meetup next week if so. If not, are we OK pulling those components in temporarily? Or should I just do the overrides I need in the context of our code with a note to migrate in the future?

Thanks a lot for getting in touch, @justinshreve!

As per p1559073709014500-slack-components, it'd be great to migrate that `TextControl` into Calypso (and thus, `calypso-ui`) :slightly_smiling_face:

**Comment by blowery** [comment]
Looks like we have some false positives from the linter for dev dependencies and a couple existing a11y problems. Thoughts?

**Comment by sirreal** [comment]
> Looks like we have some false positives from the linter for dev dependencies 

I really think we need to fork or contribute to the lint rule to better handle monorepos:

- https://github.com/benmosher/eslint-plugin-import/issues/1214
- https://github.com/benmosher/eslint-plugin-import/issues/1174
- https://github.com/benmosher/eslint-plugin-import/issues/1302

Until support is there, this shouldn't be a blocker. Lint errors serve as a reg flag for contributors and reviewers to verify dependencies are correctly declared. We can approve/comment noting the lint error is incorrect, and proceed.

> a couple existing a11y problems

If they're not too involved, would be nice to get them sorted out _before_ sharing the components.

**Comment by ockham** [comment]
> > Looks like we have some false positives from the linter for dev dependencies
> 
> I really think we need to fork or contribute to the lint rule to better handle monorepos:
> 
> * [benmosher/eslint-plugin-import#1214](https://github.com/benmosher/eslint-plugin-import/issues/1214)
> * [benmosher/eslint-plugin-import#1174](https://github.com/benmosher/eslint-plugin-import/issues/1174)
> * [benmosher/eslint-plugin-import#1302](https://github.com/benmosher/eslint-plugin-import/issues/1302)
> 
> Until support is there, this shouldn't be a blocker. Lint errors serve as a reg flag for contributors and reviewers to verify dependencies are correctly declared. We can approve/comment noting the lint error is incorrect, and proceed.

:+1: 

> > a couple existing a11y problems
> 
> If they're not too involved, would be nice to get them sorted out _before_ sharing the components.

1eac061

Ready for another look!

File: packages/calypso-build/bin/copy-styles.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const copyOptions = {
+	overwrite: true,
+	filter: '**/*.scss',
+	concurrency: 127,
+	debug: true,
+};
+
+rcopy( inputDir, outputDirEsm, copyOptions )
+	.then( results => {
+		console.log( 'copied %d files', results.length );
```
Comment (blowery) [inline packages/calypso-build/bin/copy-styles.js]: i think we can drop all the logging here. it gets pretty noisy.

Comment (ockham) [reply packages/calypso-build/bin/copy-styles.js]: Indeed:

image

Looks like `rcopy` is noisy enough on its own:

image

Comment (ockham) [reply packages/calypso-build/bin/copy-styles.js]: Disabling `rcopy`'s `debug` flag makes more of a difference:

image

Comment (blowery) [reply packages/calypso-build/bin/copy-styles.js]: nice

**Review by blowery — APPROVED** [review]
👍
